LoRaEngine/LoraKeysManagerFacade/SyncDevAddrCache.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Azure.WebJobs.Host;
+    using Microsoft.Extensions.Logging;
+
+    public class SyncDevAddrCache
+    {
+        private LoRaDevAddrCache loRaDevAddrCache;
+        private RegistryManager registryManager;
+
+        public SyncDevAddrCache(LoRaDevAddrCache loRaDevAddrCache, RegistryManager registryManager)
+        {
+            this.loRaDevAddrCache = loRaDevAddrCache;
+            this.registryManager = registryManager;
+        }
+
+        [FunctionName("SyncDevAddrCache")]
+        public async Task Run([TimerTrigger("0 */5 * * * *")]TimerInfo myTimer, ILogger log)
```

Comment:
Should we add tests to the function to ensure the device cache:
-  is being cleared 
- operation is waited
- exception are raised

?

Reply:
I think a lot of tests regarding cache function are already present [here](https://github.com/Azure/iotedge-lorawan-starterkit/blob/fdbc3c17bf5bb0805bbf6578d8bd6d60f229395b/LoRaEngine/test/LoraKeysManagerFacade.Test/DevAddrCacheTest.cs).
Regarding the operation is waited, what exactly do you mean? And what exception do you have in mind?

Reply:
If loRaDevAddrCache.PerformNeededSyncs throws an exception